test/utils/readRowsImpl.ts

```diff
@@ -231,60 +307,7 @@ export function readRowsImpl(
     });
 
     let chunksSent = 0;
-    let keyFromRequestClosed: string | undefined;
-    if (
-      stream?.request?.rows?.rowRanges &&
-      stream?.request?.rows?.rowRanges[0] &&
-      stream?.request?.rows?.rowRanges[0]?.startKeyClosed?.toString()
-    ) {
-      keyFromRequestClosed =
-        stream?.request?.rows?.rowRanges[0]?.startKeyClosed?.toString();
-    }
-    let keyFromRequestOpen: string | undefined;
-    if (
-      stream?.request?.rows?.rowRanges &&
-      stream?.request?.rows?.rowRanges[0] &&
-      stream?.request?.rows?.rowRanges[0]?.startKeyOpen?.toString()
-    ) {
-      keyFromRequestOpen =
-        stream?.request?.rows?.rowRanges[0]?.startKeyOpen?.toString();
-    }
-    let keyToRequestClosed: string | undefined;
-    if (
-      stream?.request?.rows?.rowRanges &&
-      stream?.request?.rows?.rowRanges[0] &&
-      stream?.request?.rows?.rowRanges[0]?.endKeyClosed?.toString()
-    ) {
-      keyToRequestClosed =
-        stream?.request?.rows?.rowRanges[0]?.endKeyClosed?.toString();
-    }
-    let keyToRequestOpen: string | undefined;
-    if (
-      stream?.request?.rows?.rowRanges &&
-      stream?.request?.rows?.rowRanges[0] &&
-      stream?.request?.rows?.rowRanges[0]?.endKeyOpen?.toString()
-    ) {
-      keyToRequestOpen =
-        stream?.request?.rows?.rowRanges[0]?.endKeyOpen?.toString();
-    }
-    const keyFromUsed =
-      serviceParameters.keyFrom !== undefined
-        ? serviceParameters.keyFrom
-        : keyFromRequestClosed
-          ? parseInt(keyFromRequestClosed as string)
-          : parseInt(keyFromRequestOpen as string) + 1;
-    const keyToUsed =
-      serviceParameters.keyTo !== undefined
-        ? serviceParameters.keyTo
-        : keyToRequestClosed
-          ? parseInt(keyToRequestClosed as string)
-          : parseInt(keyToRequestOpen as string) + 1;
-    const chunks = generateChunks({
-      keyFrom: keyFromUsed,
-      keyTo: keyToUsed,
-      chunkSize: serviceParameters.chunkSize,
-      valueSize: serviceParameters.valueSize,
-    });
+    const chunks = generateChunksFromRequest(stream.request, serviceParameters);
```

Comment:
This single line of code is a substitute for all of the code deleted previous to it.

Reply:
The code for each if statement now lives in `getKeyValue`. `generateChunksFromRequest` calls `generateChunks` making 2 calls to `getSelectedKey` each making 2 calls to `getKeyValue`.